### PR TITLE
[BACKLOG-29565] - PDI-CE Assembly - Migrating from deprecated old style

### PR DESCRIPTION
**Pull request by ricardosilva88**
This should be the first PR to be merged

And then:
https://github.com/pentaho/pentaho-kettle/pull/6743
https://github.com/pentaho/pentaho-karaf-ee-assembly/pull/231

CC: @Pancho7

### CODE REVIEW

File: karaf-assembly/pentaho-karaf-assembly-client/pom.xml [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+        <artifactId>karaf-maven-plugin</artifactId>
+        <version>${karaf-maven-plugin.version}</version>
+        <extensions>true</extensions>
+        <configuration>
+          <archiveTarGz>false</archiveTarGz>
+          <archiveZip>true</archiveZip>
+          <!-- override expected osgi.ee to match compilation target -->
+          <javase>1.8</javase>
+          <!-- install startup jars 'by-reference' which avoids copying -->
+          <useReferenceUrls>false</useReferenceUrls>
```
Comment (graimundo) [inline karaf-assembly/pentaho-karaf-assembly-client/pom.xml]: Why is this being used? From the code it seems that it has nothing to do with the comment.
https://github.com/apache/karaf/blob/karaf-4.2.6/tooling/karaf-maven-plugin/src/main/java/org/apache/karaf/tooling/AssemblyMojo.java#L339-L347

File: karaf-assembly/pentaho-karaf-assembly-client/src/main/resources-filtered/etc-scale/org.pentaho.features.cfg [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# Additional features to be installed at startup
+runtimeFeatures=pentaho-client,pentaho-metaverse,pentaho-dataservice,pdi-data-refinery,pdi-marketplace,community-edition
```
Comment (graimundo) [inline karaf-assembly/pentaho-karaf-assembly-client/src/main/resources-filtered/etc-scale/org.pentaho.features.cfg]: Can we use break lines here or isn't it supported? Like this:
```
runtimeFeatures=pentaho-client, \
   pentaho-metaverse, \
....
```

File: karaf-assembly/pentaho-karaf-assembly-client/src/main/resources-filtered/etc/org.apache.karaf.features.cfg [hunk 3]
```diff
@@ -23,7 +23,7 @@ respectStartLvlDuringFeatureStartup=false
...
+featuresRepositories=mvn:org.apache.karaf.features/standard/${karaf.version}/xml/features,mvn:org.apache.karaf.features/enterprise/${karaf.version}/xml/features,mvn:org.apache.cxf.karaf/apache-cxf/${cxf.version}/xml/features,mvn:org.apache.camel.karaf/apache-camel/${camel.karaf.version}/xml/features,mvn:org.hitachivantara.karaf.features/pentaho-karaf-features-standard/${project.version}/xml/features,mvn:pentaho-karaf-features/pentaho-big-data-plugin-osgi/${project.version}/xml/features,mvn:org.pentaho/pdi-marketplace/${project.version}/xml/features,mvn:org.pentaho.di.plugins/repositories-plugin/${project.version}/xml/features,mvn:org.pentaho.di.plugins/pdi-engine-configuration/${project.version}/xml/features,mvn:org.pentaho.adaptive/adaptive-execution-features/${project.version}/xml/features,mvn:org.pentaho/pdi-engine-configuration/${project.version}/xml/features,mvn:org.pentaho.di.plugins/file-open-save-plugin/${project.version}/xml/features,mvn:org.pentaho.di.plugins/file-open-save-new-plugin/${project.version}/xml/features,mvn:org.pentaho.di.plugins/get-fields-plugin/${project.version}/xml/features,mvn:org.pentaho.di.plugins/connections-plugin/${project.version}/xml/features
```
Comment (graimundo) [inline karaf-assembly/pentaho-karaf-assembly-client/src/main/resources-filtered/etc/org.apache.karaf.features.cfg]: Can we use the breakline syntax here?

File: karaf-assembly/pom.xml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>pentaho</groupId>
+    <artifactId>pentaho-karaf</artifactId>
+    <version>9.0.0.0-SNAPSHOT</version>
+  </parent>
+
+  <groupId>pentaho</groupId>
```
Comment (graimundo) [inline karaf-assembly/pom.xml]: For the groupId use `org.hitachivantara.karaf`

File: karaf-assembly/pom.xml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  <artifactId>karaf-assembly</artifactId>
+  <version>9.0.0.0-SNAPSHOT</version>
```
Comment (graimundo) [inline karaf-assembly/pom.xml]: No need to declare version

File: pentaho-karaf-assembly/src/main/filtered-resources/etc-pme/org.apache.karaf.features.cfg [hunk 6]
```diff
@@ -23,7 +23,7 @@ respectStartLvlDuringFeatureStartup=false
 #
 # Comma separated list of features repositories to register by default
 #
-featuresRepositories=mvn:org.apache.karaf.features/standard/${karaf.version}/xml/features,mvn:org.apache.karaf.features/enterprise/${karaf.version}/xml/features,mvn:org.apache.cxf.karaf/apache-cxf/${cxf.version}/xml/features,mvn:org.apache.camel.karaf/apache-camel/${camel.karaf.version}/xml/features,mvn:pentaho/pentaho-karaf-features/${project.version}/xml/standard,mvn:pentaho-karaf-features/pentaho-big-data-plugin-osgi/${project.version}/xml/features,mvn:pentaho/pentaho-dataservice-client/${project.version}/xml/features
+featuresRepositories=mvn:org.apache.karaf.features/standard/${karaf.version}/xml/features,mvn:org.apache.karaf.features/enterprise/${karaf.version}/xml/features,mvn:org.apache.cxf.karaf/apache-cxf/${cxf.version}/xml/features,mvn:org.apache.camel.karaf/apache-camel/${camel.karaf.version}/xml/features,mvn:org.hitachivantara.karaf.features/pentaho-karaf-features-standard/${project.version}/xml/features,mvn:pentaho-karaf-features/pentaho-big-data-plugin-osgi/${project.version}/xml/features,mvn:pentaho/pentaho-dataservice-client/${project.version}/xml/features
```
Comment (graimundo) [inline pentaho-karaf-assembly/src/main/filtered-resources/etc-pme/org.apache.karaf.features.cfg]: Can we use the break line syntax?

File: pentaho-karaf-assembly/src/main/filtered-resources/etc-pmr/org.apache.karaf.features.cfg [hunk 7]
```diff
@@ -22,7 +22,7 @@ respectStartLvlDuringFeatureStartup=false
 #
 # Comma separated list of features repositories to register by default
 #
-featuresRepositories=mvn:org.apache.karaf.features/standard/${karaf.version}/xml/features,mvn:org.apache.karaf.features/enterprise/${karaf.version}/xml/features,mvn:org.apache.cxf.karaf/apache-cxf/${cxf.version}/xml/features,mvn:pentaho/pentaho-karaf-features/${project.version}/xml/standard,mvn:pentaho-karaf-features/pentaho-big-data-plugin-osgi/${project.version}/xml/features
+featuresRepositories=mvn:org.apache.karaf.features/standard/${karaf.version}/xml/features,mvn:org.apache.karaf.features/enterprise/${karaf.version}/xml/features,mvn:org.apache.cxf.karaf/apache-cxf/${cxf.version}/xml/features,mvn:org.hitachivantara.karaf.features/pentaho-karaf-features-standard/${project.version}/xml/features,mvn:pentaho-karaf-features/pentaho-big-data-plugin-osgi/${project.version}/xml/features
```
Comment (graimundo) [inline pentaho-karaf-assembly/src/main/filtered-resources/etc-pmr/org.apache.karaf.features.cfg]: Can we use the breakline syntax?

File: pentaho-karaf-assembly/src/main/filtered-resources/etc-prd/org.apache.karaf.features.cfg [hunk 8]
```diff
@@ -22,7 +22,7 @@ respectStartLvlDuringFeatureStartup=false
 #
 # Comma separated list of features repositories to register by default
 #
-featuresRepositories=mvn:org.apache.karaf.features/standard/${karaf.version}/xml/features,mvn:org.apache.karaf.features/enterprise/${karaf.version}/xml/features,mvn:org.apache.cxf.karaf/apache-cxf/${cxf.version}/xml/features,mvn:org.apache.camel.karaf/apache-camel/${camel.karaf.version}/xml/features,mvn:pentaho/pentaho-karaf-features/${project.version}/xml/standard,mvn:pentaho-karaf-features/pentaho-big-data-plugin-osgi/${project.version}/xml/features,mvn:pentaho/pentaho-dataservice-client/${project.version}/xml/features
+featuresRepositories=mvn:org.apache.karaf.features/standard/${karaf.version}/xml/features,mvn:org.apache.karaf.features/enterprise/${karaf.version}/xml/features,mvn:org.apache.cxf.karaf/apache-cxf/${cxf.version}/xml/features,mvn:org.apache.camel.karaf/apache-camel/${camel.karaf.version}/xml/features,mvn:org.hitachivantara.karaf.features/pentaho-karaf-features-standard/${project.version}/xml/features,mvn:pentaho-karaf-features/pentaho-big-data-plugin-osgi/${project.version}/xml/features,mvn:pentaho/pentaho-dataservice-client/${project.version}/xml/features
```
Comment (graimundo) [inline pentaho-karaf-assembly/src/main/filtered-resources/etc-prd/org.apache.karaf.features.cfg]: Can we use the breakline syntax?

File: pentaho-karaf-assembly/src/main/filtered-resources/etc-server/org.apache.karaf.features.cfg [hunk 9]
```diff
@@ -22,7 +22,7 @@ respectStartLvlDuringFeatureStartup=false
...
+featuresRepositories=mvn:org.apache.karaf.features/standard/${karaf.version}/xml/features,mvn:org.apache.karaf.features/enterprise/${karaf.version}/xml/features,mvn:org.apache.camel.karaf/apache-camel/${camel.karaf.version}/xml/features,mvn:org.hitachivantara.karaf.features/pentaho-karaf-features-standard/${project.version}/xml/features,mvn:org.hitachivantara.karaf.features/pentaho-karaf-features-server/${project.version}/xml/features,mvn:org.apache.activemq/activemq-karaf/${activemq.version}/xml/features,mvn:pentaho-karaf-features/pentaho-big-data-plugin-osgi/${project.version}/xml/features,mvn:org.apache.cxf.karaf/apache-cxf/${cxf.version}/xml/features,mvn:org.pentaho/pentaho-marketplace/${project.version}/xml/features,mvn:org.pentaho.di.plugins/pdi-engine-configuration/${project.version}/xml/features,mvn:org.pentaho.adaptive/adaptive-execution-features/${project.version}/xml/features,mvn:org.pentaho/pdi-engine-configuration/${project.version}/xml/features,mvn:pentaho/pentaho-dataservice-legacy/${project.version}/xml/features,mvn:org.pentaho.di.plugins/get-fields-plugin/${project.version}/xml/features
```
Comment (graimundo) [inline pentaho-karaf-assembly/src/main/filtered-resources/etc-server/org.apache.karaf.features.cfg]: Can we use the breakline syntax?

File: karaf-assembly/pentaho-karaf-assembly-client/pom.xml (hunk 1 above)
Comment (ricardosilva88) [reply karaf-assembly/pentaho-karaf-assembly-client/pom.xml]: useReferenceUrls and ignoreDependencyFlag are already false by default

**Pull request merged**